### PR TITLE
Add native JavaScript(TypeScript) to replace translation.py

### PR DESCRIPTION
**Pull request by huan**
In our [translation](https://github.com/tensorflow/tfjs-examples/tree/master/translation) example, we use a python script to generate the model, train, and save the model to file for future use.

I think it will be more clear to use a native JavaScript script to do this task, so I rewrite it from Python to TypeScript.

Train the model and save it to file in JavaScript demostrated as the following:

```sh
$ yarn train /dist/fra.txt 
yarn run v1.12.3
$ node -r ts-node/register translation.ts --artifacts_dir dist/resources /home/zixia/git/javascript-concist-chit-chat/data/fra.txt
2019-03-07 15:02:07.198157: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
Number of samples: 10000
Number of unique input tokens: 69
Number of unique output tokens: 93
Max sequence length for inputs: 16
Max sequence length for outputs: 59
Saved metadata at:  dist/resources/metadata.json
Orthogonal initializer is being called on a matrix with more than 2000 (262144) elements: Slowness may result.
Orthogonal initializer is being called on a matrix with more than 2000 (262144) elements: Slowness may result.
Epoch 1 / 20
eta=0.0 =============================================================================================================================> 
59356ms 7419us/step - loss=1.11 val_loss=1.10 
Epoch 2 / 20
eta=0.0 =============================================================================================================================> 
60262ms 7533us/step - loss=0.944 val_loss=1.04 
Epoch 3 / 20
eta=0.0 =============================================================================================================================> 

...
...
...

63610ms 7951us/step - loss=0.529 val_loss=0.631 
Epoch 17 / 20
eta=0.0 =============================================================================================================================> 
64396ms 8050us/step - loss=0.521 val_loss=0.618 
Epoch 18 / 20
eta=0.0 =============================================================================================================================> 
61433ms 7679us/step - loss=0.510 val_loss=0.611 
Epoch 19 / 20
eta=0.0 =============================================================================================================================> 
61209ms 7651us/step - loss=0.499 val_loss=0.600 
Epoch 20 / 20
eta=0.0 =============================================================================================================================> 
61208ms 7651us/step - loss=0.492 val_loss=0.590 
```

Cheers.

CC @wangtz

BTW: I'd like to suggest let's open the ISSUE entry for this repo because the examples might have many problems need to be discussed.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-examples/243)

### CODE REVIEW

File: translation/translation.ts [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+    decoder_inputs: Instance of `keras.Input`, symbolic tensor as input to
+      the decoder LSTM.
+    decoder_lstm: `keras.Layer` instance, the decoder LSTM.
+    decoder_dense: `keras.Layer` instance, the Dense layer in the decoder.
+    model: `keras.Model` instance, the entire translation model that can be
+      used in training.
+  */
+  // Define an input sequence and process it.
+  const encoder_inputs = tf.layers.input({
+    shape: [null, num_encoder_tokens] as number[],
```
Comment (huan) [inline translation/translation.ts]: See: https://github.com/tensorflow/tfjs-layers/pull/492#issuecomment-470785861
@davidsoergel @caisq

**Comment by nsthorat** [comment]
Once https://github.com/tensorflow/tfjs-node/pull/209 goes in, we'll have a 1.0 for node.js.

**Comment by huan** [comment]
@nsthorat That's right. I found that the tfjs-node does not work with the current tfjs 1.0. I'll upgrade this example to TFJS 1.0 when they getting compatible stably.

@caisq Thanks for the review, I'll follow your reviews and modify this PR as you requested later.

**Comment by huan** [comment]
Hi @caisq, I had modified my PR as your request, except that I did not add the explicitly type for the return of function because they can be referenced automatically from the return statement inside the function.

Please let me know if there need any other modifications, thanks.

**Comment by huan** [comment]
@caisq Thanks for the additional reviews!

I had followed your reviews and updated the PR as well.

**Comment by huan** [comment]
@caisq Done.

**Pull request merged**